File: core/src/main/java/javabot/web/views/AdminIndexView.java
package javabot.web.views;

import com.google.inject.Injector;
import javabot.dao.AdminDao;
import javabot.model.Admin;

import javax.inject.Inject;
import javax.servlet.http.HttpServletRequest;
import java.util.List;

public class AdminIndexView extends MainView {

    @Inject
    private AdminDao adminDao;

    public AdminIndexView(final Injector injector, final HttpServletRequest request) {
        super(injector, request);
    }

    public List<Admin> getAdmins() {
        return adminDao.findAll();
    }

    @Override
    public String getChildView() {
        return "admin/index.ftl";
    }
}

File: src/main/kotlin/javabot/web/views/AdminIndexView.kt
package javabot.web.views

import jakarta.inject.Inject
import jakarta.servlet.http.HttpServletRequest
import javabot.dao.AdminDao
import javabot.dao.ApiDao
import javabot.dao.ChannelDao
import javabot.dao.FactoidDao
import javabot.model.Admin

class AdminIndexView
@Inject
constructor(
    adminDao: AdminDao,
    channelDao: ChannelDao,
    factoidDao: FactoidDao,
    apiDao: ApiDao,
    request: HttpServletRequest,
    val current: Admin,
    val editing: Admin?
) : MainView(adminDao, channelDao, factoidDao, apiDao, request) {

    fun getAdmins(): List<Admin> {
        return adminDao.findAll()
    }

    override fun getChildView(): String {
        return "admin/index.ftl"
    }
}
